### ISSUE TITLE
The tests are absurdly slow

### ISSUE THREAD

**Issue by FlutterIssues** [flutter/flutter#161]
<a href="https://github.com/Hixie"><img src="https://avatars.githubusercontent.com/u/551196?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [Hixie](https://github.com/Hixie)**
_Tuesday Sep 15, 2015 at 19:37 GMT_
_Originally opened as https://github.com/flutter/engine/issues/1172_

----

The tests take too long to run. I don't understand why; they should take mere milliseconds.

**Comment by FlutterIssues** [comment]
<a href="https://github.com/abarth"><img src="https://avatars.githubusercontent.com/u/112007?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [abarth](https://github.com/abarth)**
_Tuesday Sep 15, 2015 at 19:57 GMT_

----

How long do they take to run on your machine?

**Comment by FlutterIssues** [comment]
<a href="https://github.com/Hixie"><img src="https://avatars.githubusercontent.com/u/551196?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [Hixie](https://github.com/Hixie)**
_Tuesday Sep 15, 2015 at 20:16 GMT_

----

43s wall-clock (1m18s user, presumably across two CPUs?).
For comparison, the analyzer takes 6 seconds (which is also too long), and telling the device to start the app takes around 2 seconds (which is also too long).

**Comment by FlutterIssues** [comment]
<a href="https://github.com/abarth"><img src="https://avatars.githubusercontent.com/u/112007?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [abarth](https://github.com/abarth)**
_Tuesday Sep 15, 2015 at 20:25 GMT_

----

Sounds like the same issue @apwilson was having with the test harness crawling symlinks.

**Comment by FlutterIssues** [comment]
<a href="https://github.com/Hixie"><img src="https://avatars.githubusercontent.com/u/551196?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [Hixie](https://github.com/Hixie)**
_Tuesday Sep 15, 2015 at 20:44 GMT_

----

yeah, though i wasn't able to figure out which version of the test package it's using so i could change it. :-/

**Comment by FlutterIssues** [comment]
<a href="https://github.com/Hixie"><img src="https://avatars.githubusercontent.com/u/551196?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [Hixie](https://github.com/Hixie)**
_Wednesday Sep 16, 2015 at 19:02 GMT_

----

Oh hey I did a gclient sync and now it does pick up my hacked copy of test. Now it runs in 6 seconds!

**Comment by FlutterIssues** [comment]
<a href="https://github.com/apwilson"><img src="https://avatars.githubusercontent.com/u/142874?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [apwilson](https://github.com/apwilson)**
_Thursday Sep 17, 2015 at 20:29 GMT_

----

Awesome!

Just for reference, the workaround is this:  https://github.com/apwilson/test/commit/368cf30a5d154d971dba72302f8c3b32a83b2ae7

**Comment by FlutterIssues** [comment]
<a href="https://github.com/apwilson"><img src="https://avatars.githubusercontent.com/u/142874?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [apwilson](https://github.com/apwilson)**
_Thursday Sep 17, 2015 at 20:31 GMT_

----

And for reasons why dart/test don't want to implement that fix, see: https://github.com/dart-lang/test/issues/337

TLDR: This issue will fix itself in a few months when the dart packages link goes away.

**Comment by abarth** [comment]
@Hixie I think this is fixed now that `flutter test` does it's own walk to find the files.
